src/server/bitops_family.cc

```diff
@@ -1176,7 +1182,15 @@ void BitOp(CmdArgList args, ConnectionContext* cntx) {
         }
 
         if (shard->journal()) {
-          RecordJournal(t->GetOpArgs(shard), "SET", {dest_key, op_result});
+          if (op_result.empty()) {
```

Comment:
also if find_res == OpStatus::OK

Reply:
my question here would be if the result is not ok why do we `RecordJournal` ?

Reply:
we should not

Reply:
we had another bug then :stuck_out_tongue:

Reply:
setup master and replica and then:

1. on master do: hset foo bar val -> bitop not foo bar -> get foo returns WRONGTYPE 
2. on replica do GET foo returns ""

Now state_of(master) != state_of(replica)

Reply:
I added this on the issue + supplied the fix + added a replication test case so we don't break it again